packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2665,6 +2666,77 @@ class SizedOverflowBox extends SingleChildRenderObjectWidget {
 /// needed, prefer removing the widget from the tree entirely rather than
 /// keeping it alive in an [Offstage] subtree.
 ///
+/// {@tool dartpad --template=stateful_widget_material_ticker}
+///
+/// This example shows a [FlutterLogo] widget when the `_offstage` member field
+/// is false, and hides it without any room in the parent when it is true. The
+/// [Size] of the logo without bringing it on screen will be measured.
+///
+/// ```dart
+/// class OffstageLogo extends StatefulWidget {
+///   @override
+///   _OffstageLogoState createState() => _OffstageLogoState();
+/// }
+///
+/// class _OffstageLogoState extends State<OffstageLogo> {
+///   GlobalKey _key = GlobalKey();
+///   bool _offstage = true;
+///   Size _invisibleLogoSize;
+///
+///   void _changeOffstageValue() {
+///     setState(() {
+///       _offstage = !_offstage;
+///       _invisibleLogoSize = null;
+///     });
+///   }
+///
+///   void _getInvisibleLogoSize() {
+///     setState(() {
+///       final RenderBox renderLogo = _key.currentContext.findRenderObject();
+///       final sizeLogo = renderLogo.size;
+///       _invisibleLogoSize = sizeLogo;
+///     });
+///   }
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return new Scaffold(
+///       body: Center(
+///         child: Column(
+///           mainAxisAlignment: MainAxisAlignment.center,
+///           children: <Widget>[
+///             Offstage(
+///               offstage: _offstage,
+///               child: FlutterLogo(
+///                 key: _key,
+///                 size: 150.0,
+///               ),
+///             ),
+///             RaisedButton(
+///               child: Text('Change Offstage Value'),
+///               onPressed: () { _changeOffstageValue(); },
+///             ),
+///             Offstage(
+///               offstage: !_offstage,
+///               child: Column(
+///                 children: <Widget>[
+///                   RaisedButton(
+///                     child: const Text('Get Size Of The Invisible Logo'),
+///                     onPressed: () { _getInvisibleLogoSize(); },
+///                   ),
+///                   Text('${_invisibleLogoSize ?? ''}'),
+///                 ],
+///               ),
+///             ),
```

Comment:
This particular widget doesn't have to be offstage, does it? My assumption is that you're trying to demonstrate that we can get the size of `FlutterLogo`, even when it is offstage. 

I think a better example could be something like so:
- We have an Offstage widget, much like yours, containing a widget (`FlutterLogo`) of a certain size.
- Have a button to toggle offstage, just to demonstrate what happens visually when offstage is set to true/false.
- Have a text widget that displays the following:
```
if offstage is false, show "FlutterLogo widget is not currently offstage"
if offstage is true, show "FlutterLogo widget is offstage. It is laid out in the widget tree, but occupies no space visually, cannot be hit tested, and is not painted. Yet, we are able to get its size: ${logoSize}"
```